public/app/features/datasources/state/selectors.ts

```diff
@@ -4,9 +4,15 @@ import { DataSourcesState } from 'app/types/datasources';
 export const getDataSources = (state: DataSourcesState) => {
   const regex = new RegExp(state.searchQuery, 'i');
 
-  return state.dataSources.filter((dataSource: DataSourceSettings) => {
+  const filteredDataSources = state.dataSources.filter((dataSource: DataSourceSettings) => {
     return regex.test(dataSource.name) || regex.test(dataSource.database) || regex.test(dataSource.type);
   });
+
+  if (state.isSortAscending) {
+    return filteredDataSources.sort((a, b) => a.name.localeCompare(b.name));
+  } else {
+    return filteredDataSources.sort((a, b) => b.name.localeCompare(a.name));
+  }
```

Comment:
Maybe (just a little bit more dry, not 100% sure it worth it)?
```javascript
return filteredDataSources.sort((a, b) => 
  state.isSortAscending ? 
    a.name.localeCompare(b.name) :
    b.name.localeCompare(a.name)
  );
```

Reply:
Done